ForumProject/communications/views.py

```diff
@@ -1,3 +1,50 @@
-from django.shortcuts import render
+from django.contrib.auth import get_user_model
+from django.contrib.auth.decorators import login_required
+from django.shortcuts import render, get_object_or_404
+from .serializers import RoomSerializer, MessageSerializer
+from .models import Room, Message
 
-# Create your views here.
+from rest_framework import generics
+from .models import Room, Message
+
+User = get_user_model()
+
+
+@login_required
+def index_view(request):
+    users = User.objects.exclude(id=request.user.id)
+    rooms_list = Room.objects.filter(name__contains=str(request.user.id))
+    return render(request, 'index.html', {
+        'rooms': rooms_list, 'users': users
+    })
+
+
+@login_required
+def room_view(request, user_id):
+    user = get_object_or_404(User, id=user_id)
+    if request.user.id > user.id:
+        thread_name = f'chat_{request.user.id}_{user.id}'
+    else:
+        thread_name = f'chat_{user.id}_{request.user.id}'
+    chat_room, created = Room.objects.get_or_create(name=thread_name)
+
+    messages = Message.objects.filter(room=chat_room)
+    users_messages = '\n'.join([str(message) for message in messages])+'\n'
```

Comment:
Using `'\n'.join([str(message) for message in messages])` to pass messages to the template is unusual and might not be the best way to handle message formatting. It's better to handle such formatting in the template itself or use JSON.

Reply:
done